### PR TITLE
[tests-only] Added GUI test scenario to check local sync and default sync options

### PR DESCRIPTION
**Pull request by Salipa-Gurung**
### Description
This pr adds test scenario related to:
* checking availability of local sync directory option in advanced configuration.
* checking  default sync option
### Related Issue 
https://github.com/owncloud/QA/issues/789

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-ocis</strong> https://drone.owncloud.com/owncloud/client/14136/7/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/14136/ocis/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/14136/ocis/guiReportUpload/serverlog.log

Screenshots:
- [Choosing_local_download_directory_in_advanced_configuration.png](https://cache.owncloud.com/public/owncloud/client/14136/ocis/guiReportUpload/screenshots/Choosing_local_download_directory_in_advanced_configuration.png)

</details>

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests-oc10</strong> https://drone.owncloud.com/owncloud/client/14136/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/14136/oc10/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/14136/oc10/guiReportUpload/serverlog.log

Screenshots:
- [Verify_one_empty_folder_with_a_length_longer_than_the_allowed_limit_will_not_be_synced.png](https://cache.owncloud.com/public/owncloud/client/14136/oc10/guiReportUpload/screenshots/Verify_one_empty_folder_with_a_length_longer_than_the_allowed_limit_will_not_be_synced.png)

</details>

File: test/gui/tst_syncing/test.feature [hunk 1]
```diff
@@ -353,4 +353,13 @@ Feature: Syncing files
...
+        And as user "Alice" folder "folder2" should contain "500" items on the server
+
+
+    Scenario: Choosing local download directory in advanced configuration
+        Given the user has started the client
+        When the user adds the following account information:
+            | server   | %local_server% |
+            | user     | Alice          |
+            | password | 1234           |
+        Then local folder selection option should be visible
```
Comment (amrita-shrestha) [inline test/gui/tst_syncing/test.feature:365]: choosing local directory step should be in When step

File: test/gui/tst_syncing/test.feature [hunk 2]
```diff
@@ -353,4 +353,13 @@ Feature: Syncing files
...
\ No newline at end of file
+        And as user "Alice" folder "folder2" should contain "500" items on the server
+
+
+    Scenario: Choosing local download directory in advanced configuration
+        Given the user has started the client
+        When the user adds the following account information:
+            | server   | %local_server% |
+            | user     | Alice          |
+            | password | 1234           |
```
Comment (amrita-shrestha) [inline test/gui/tst_syncing/test.feature:364]: this step should already exist. Or it was removed in refactoring. @saw-jan

**Review by amrita-shrestha — CHANGES_REQUESTED** [review]

File: test/gui/shared/scripts/pageObjects/AccountConnectionWizard.py [hunk 3]
```diff
@@ -125,6 +125,12 @@ class AccountConnectionWizard:
         "unnamed": 1,
         "visible": 1,
     }
+    LOCAL_DOWNLOAD_DIRECTORY = {
```
Comment (saw-jan) [inline test/gui/shared/scripts/pageObjects/AccountConnectionWizard.py:128]:
```suggestion
    LOCAL_SYNC_DIRECTORY = {
```

File: test/gui/shared/scripts/pageObjects/AccountConnectionWizard.py [hunk 4]
```diff
@@ -281,3 +285,19 @@ def isCredentialWindowVisible():
...
         return visible
+
+    @staticmethod
+    def selectAdvancedConfig():
+        squish.waitForObject(
+            AccountConnectionWizard.ADVANCED_CONFIGURATION_CHECKBOX
+        ).setChecked(True)
+
+    @staticmethod
+    def isLocalDownloadDirVisible():
```
Comment (saw-jan) [inline test/gui/shared/scripts/pageObjects/AccountConnectionWizard.py:296]:
```suggestion
    def isLocalSyncDirVisible():
```

**Review by saw-jan — COMMENTED** [review]
Sync would be appropriate here

File: test/gui/tst_syncing/test.feature (hunk 1 above)
Comment (saw-jan) [reply test/gui/tst_syncing/test.feature:365]: The step is not to choose the folder but assert that the user can see the option to select the sync folder

File: test/gui/tst_syncing/test.feature (hunk 2 above)
Comment (saw-jan) [reply test/gui/tst_syncing/test.feature:364]: We have `Given` step but not `When` I guess

File: test/gui/shared/steps/account_context.py [hunk 5]
```diff
@@ -196,3 +196,22 @@ def step(context, action):
...
+        AccountConnectionWizard.acceptCertificate()
+    AccountConnectionWizard.addUserCreds(
+        account_details['user'], account_details['password']
+    )
+
+
+@Then("local folder selection option should be visible")
+def step(context):
+    AccountConnectionWizard.selectAdvancedConfig()
+    AccountConnectionWizard.isLocalSyncDirVisible()
```
Comment (saw-jan) [inline test/gui/shared/steps/account_context.py:217]: Add the assertion in the Then step
```suggestion
@Then("local folder selection option should be visible")
def step(context):
    visible = AccountConnectionWizard.isLocalSyncDirVisible()
    test.compare(visible, True, <some-message>)
```

File: test/gui/tst_syncing/test.feature (hunk 1 above)
Comment (amrita-shrestha) [reply test/gui/tst_syncing/test.feature]: I think scenario title says `Choosing local download directory in advanced configuration` so i got confused

File: test/gui/tst_syncing/test.feature [hunk 6]
```diff
@@ -353,4 +353,15 @@ Feature: Syncing files
         Then as "Alice" folder "folder1" should exist on the server
         And as user "Alice" folder "folder1" should contain "500" items on the server
         And as "Alice" folder "folder2" should exist on the server
-        And as user "Alice" folder "folder2" should contain "500" items on the server
\ No newline at end of file
+        And as user "Alice" folder "folder2" should contain "500" items on the server
+
+
+    Scenario: Default options in advanced configuration
```
Comment (amrita-shrestha) [inline test/gui/tst_syncing/test.feature:359]: the scenario should open up/ give an idea about what scenario tests.
I think it can be better

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by amrita-shrestha — APPROVED** [review]

**Pull request merged**